browser/src/backendMachine.js

```diff
@@ -0,0 +1,152 @@
+import { Machine, assign, sendParent } from 'xstate';
+import $ from 'jquery';
+
+const getModel = () => window.model;
+
+const backendMachine = Machine(
```

Comment:
<img width="869" alt="Screen Shot 2021-02-16 at 2 17 52 PM" src="https://user-images.githubusercontent.com/40068392/108129705-6300c480-7063-11eb-83ab-cec4b63673db.png">

Here is a visualization of this state machine.

- this machine is [invoked](https://xstate.js.org/docs/guides/communication.html) by the main deepcell label statechart
- the idle state means that we are not waiting a response & it can receive messages to call specific routes, like edit or undo
- When we receive a message like EDIT or UNDO, we leave the idle state & enter the state to call the routes and wait for a response
- When leaving the idle state, we send a LOADING message back to the parent so the parent can update its behavior
- Once we receive a response, we transiently enter the loaded state to handle the response & send a LOADED response back to the parent

Reply:
This state machine also handles errors from these routes, recording the error and sending an ERROR message, but the app doesn't currently do anything with these messages.